src/Streamly/Internal/Data/Array/Prim/TypesInclude.hs

```diff
@@ -28,6 +26,10 @@ import qualified Streamly.Internal.Data.Stream.StreamK as K
 import GHC.Exts hiding (fromListN, fromList, toList)
 import Prelude hiding (length, unlines, foldr)
 
+#if __GLASGOW_HASKELL__ < 808
+import Data.Semigroup (Semigroup(..))
+#endif
```

Comment:
Why move this? It creates yet another grouping for imports. I think its ok to keep the conditional imports wherever they fall in the sort order.